### PR TITLE
sdk_key and environment_key support 

### PR DESCRIPTION
**Pull request by The-inside-man**
Summary
-------

-  Added test cases to improve coverage
-  Fixed spelling mistakes in environment_key
-  sdk_key and environment_key support added for optimizely_config

Test plan
---------

-  Compatibility test suite and compliance tests

Issues
------

-  OASIS-7757

### CODE REVIEW

File: tests/test_optimizely.py [hunk 1]
```diff
@@ -266,6 +266,19 @@ def test_init__sdk_key_and_datafile_access_token(self):
 
         self.assertIs(type(opt_obj.config_manager), config_manager.AuthDatafilePollingConfigManager)
 
+    def test_init__invalid_default_decide_options(self):
```
Comment (jaeopt) [inline tests/test_optimizely.py:269]: what about moving this test to "test_user_context" along with other tests for defaut_decide_options?

**Review by jaeopt — APPROVED** [review]
LGTM with a suggestion

File: tests/test_optimizely.py (hunk 1 above)
Comment (The-inside-man) [reply tests/test_optimizely.py:269]: Sure, that makes sense. I can include that in my next PR I am working on for the Events and Attributes addition so that should be coming shortly.

File: optimizely/optimizely_config.py [hunk 2]
```diff
@@ -17,11 +17,13 @@
 
 
 class OptimizelyConfig(object):
-    def __init__(self, revision, experiments_map, features_map, datafile=None):
+    def __init__(self, revision, experiments_map, features_map, datafile=None, sdk_key=None, environment_key=None):
```
Comment (msohailhussain) [inline optimizely/optimizely_config.py:20]: update copyright please. :)

**Review by msohailhussain — CHANGES_REQUESTED** [review]
please add some more unit tests, asserting sdkKey and environmentKey other than None values.

File: tests/test_optimizely.py (hunk 1 above)
Comment (The-inside-man) [reply tests/test_optimizely.py]: Moved over decide tests to test_user_context.

File: optimizely/optimizely_config.py (hunk 2 above)
Comment (The-inside-man) [reply optimizely/optimizely_config.py:20]: updated copyright for files.

**Review by msohailhussain — APPROVED** [review]
lgtm
